src/backend/distributed/executor/multi_utility.c

```diff
@@ -1255,7 +1255,13 @@ ExecuteDistributedDDLCommand(Oid relationId, const char *ddlCommandString,
 {
 	bool executionOK = false;
 
-	PreventTransactionChain(isTopLevel, "distributed DDL commands");
+	if (IsInTransactionChain(isTopLevel) && XactOperations != 0)
```

Comment:
Well, reading the code makes it seem like these are 2 independent conditions, but there is no way XactModification > XACT_MODIFICATION_NONE could be true without  IsInTransactionChain(isTopLevel) being true as far as I can see. We could put an Assert(IsInTransactionChain(isTopLevel)) within the if-block to clarify that.